### ISSUE TITLE
pylint false positive - Instance of 'str' has no 'value' member (no-member)

### ISSUE THREAD

**Issue by mdeshmu** [pylint-dev/pylint#8897]
### Bug description

For the following piece of code pylint gives the error: `E1101: Instance of 'str' has no 'value' member (no-member)` in python 3.9 and 3.10 but doesn't gives this error in python 3.11, While the code itself works fine. It seems like a false positive for us. 

```
import sys
from enum import Enum

if sys.version_info >= (3, 11):
    from enum import StrEnum
else:

    class StrEnum(str, Enum):
        pass


class FilterOperator(StrEnum):
    IN = "IN"


print(FilterOperator.IN.value)

```

### Configuration

https://github.com/apache/superset/blob/master/.pylintrc

### Command used

```shell
pylint -E enum_test.py
```

### Pylint output

```shell
E1101: Instance of 'str' has no 'value' member (no-member)
```

### Expected behavior

pylint should be able to recognize `value` retrieval for the Enum member.

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.9.5 or Python 3.10.6
```

### OS / Environment

Linux Based Operating Systems

**Comment by mdeshmu** [comment]
which recent pylint version was this issue fixed in?

**Comment by Pierre-Sassoulas** [comment]
According to the milestone assigned to this issue, 2.17.6 and above.
